File: string/1790_check_if_one_string_swap_can_make_strings_equal/1790_check_if_one_string_swap_can_make_strings_equal.go
package main

import "fmt"

func areAlmostEqual(s1 string, s2 string) bool {
	if s1 == s2 {
		return true
	}
	if len(s1) != len(s2) {
		return false
	}
	num := 0
	dif := []int{}
	for i := 0; i < len(s1); i++ {
		if s1[i] != s2[i] {
			num += 1
			dif = append(dif, i)
		}
	}
	if num != 2 {
		return false
	}
	if s1[dif[0]] == s2[dif[1]] && s1[dif[1]] == s2[dif[0]] {
		return true
	}
	return false
}

func main() {
	s1 := "bank"
	s2 := "kanb"
	res := areAlmostEqual(s1, s2)
	fmt.Println(res)
}

File: string/1790_check_if_one_string_swap_can_make_strings_equal/1790_check_if_one_string_swap_can_make_strings_equal.py
from typing import List

class Solution:
    def areAlmostEqual(self, s1: str, s2: str) -> bool:
        test = []
        if s1 == s2:
            return True
        for i in range(len(s1)):
            if s1[i] != s2[i]:
                test.append(i)
            else:
                pass
        if len(test) != 2:
            return False
        else:
            return(s1[test[0]] == s2[test[1]] and s1[test[1]] == s2[test[0]]) 



s1 = "bank"
s2 = "kanb"
res = Solution().areAlmostEqual(s1, s2)
print(res)